### DISCUSSION TITLE
MacOS might actually happen

### DISCUSSION THREAD

**Discussion by Jeffser** [Jeffser/Alpaca#413] [Announcements]
![Screenshot 2024-12-22 at 20 31 30](https://github.com/user-attachments/assets/7d16f075-4038-4aa9-b8b0-4210319fb813)

So I did a thing

**Comment by Jeffser** [comment]
![Screenshot 2024-12-22 at 22 32 02](https://github.com/user-attachments/assets/060aa627-36cd-457b-818b-fd434990de58)

Update: Icons work, it connects to remote Ollama instance.

Font is small and it uses Apple's font instead of Cantarell, which is good I think

**Reply by Jeffser** [reply]
I can't figure out how to move the titlebar buttons to the MacOS position, and I have no idea how to package this beast

**Reply by olumolu** [reply]
I like it and does not matter as such...i think for now 3.0.0 this can be released. Other issue here and there can be fixed later.

**Comment by Jeffser** [comment]
![Screenshot 2024-12-23 at 19 04 25](https://github.com/user-attachments/assets/cc9a1329-cd7d-4b05-8f73-6f5c2422340a)

After a lot of work I made custom titlebar buttons, ported stuff to the way MacOS work and packaged it as an .app file, now all that's left is to do a .dmg so it can be installed like any other MacOS app outside the appstore

**Reply by fnoorden-vs** [reply]
I like it, did you ever release it?

**Reply by Jeffser** [reply]
well it did work but it was more of a "it works in my system", I'm still not sure how MacOS packaging works but I'm not opposed to releasing it again properly (maybe as a homebrew)

**Reply by fnoorden-vs** [reply]
Homebrew would be a solution. Nevertheless MacOS packaging is the way to go for Mac. Can I help you and describe the steps of DMG package creation?

**Reply by Jeffser** [reply]
I kinda know how to do it the problem is that I don't know how to include GTK4, Adwaita and other libraries with the package, if they are installed with homebrew (which they were on my system) it would work

**Reply by rolandixor** [reply]
> I kinda know how to do it the problem is that I don't know how to include GTK4, Adwaita and other libraries with the package, if they are installed with homebrew (which they were on my system) it would work

Would really love if you did this 🥺

Not for me personally, but I know someone would benefit from running Alpaca on OSX. Would gladly provide feedback as well
